swift/Samples/SplitScreenContainer/Sources/SplitScreenContainerViewController.swift

```diff
@@ -0,0 +1,121 @@
+        separatorView.translatesAutoresizingMaskIntoConstraints = false
+        rightContainerView.translatesAutoresizingMaskIntoConstraints = false
+
+        view.addSubview(leftContainerView)
+        view.addSubview(separatorView)
+        view.addSubview(rightContainerView)
+
+        NSLayoutConstraint.activate([
+            leftContainerView.topAnchor.constraint(equalTo: view.topAnchor),
+            leftContainerView.leftAnchor.constraint(equalTo: view.leftAnchor),
+            leftContainerView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            leftContainerView.rightAnchor.constraint(equalTo: separatorView.leftAnchor),
+
+            separatorView.topAnchor.constraint(equalTo: view.topAnchor),
+            separatorView.widthAnchor.constraint(equalToConstant: 1.0),
+            separatorView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+
+            rightContainerView.topAnchor.constraint(equalTo: view.topAnchor),
+            rightContainerView.leftAnchor.constraint(equalTo: separatorView.rightAnchor),
+            rightContainerView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            rightContainerView.rightAnchor.constraint(equalTo: view.rightAnchor),
+        ])
+        
+        updateWidthConstraints(ratio: screen.ratio, animated: false)
+        
+        self.leftContentViewController = embed(screen.leftScreen, in: leftContainerView)
+        self.rightContentViewController = embed(screen.rightScreen, in: rightContainerView)
+    }
+    
+    private func embed<ScreenType: Screen>(_ screen: ScreenType, in containerView: UIView) -> ScreenViewController<ScreenType> {
+        let viewController = viewRegistry.provideView(for: screen)
+        
+        addChild(viewController)
+        
+        viewController.view.translatesAutoresizingMaskIntoConstraints = false
+        containerView.addSubview(viewController.view)
+        
+        NSLayoutConstraint.activate([
+            viewController.view.topAnchor.constraint(equalTo: containerView.topAnchor),
+            viewController.view.rightAnchor.constraint(equalTo: containerView.rightAnchor),
+            viewController.view.bottomAnchor.constraint(equalTo: containerView.bottomAnchor),
+            viewController.view.leftAnchor.constraint(equalTo: containerView.leftAnchor),
+            ])
+        
+        viewController.didMove(toParent: self)
+        
+        return viewController
+    }
+    
+    private func updateWidthConstraints(ratio: CGFloat, animated: Bool) {
+        guard animated else {
+            self.leftContainerViewWidthConstraint = self.leftContainerView.widthAnchor.constraint(equalTo: self.view.widthAnchor, multiplier: ratio)
+            self.leftContainerViewWidthConstraint?.isActive = true
+            return
+        }
+        
+        view.layoutIfNeeded()
+        UIView.animate(withDuration: 0.2) {
+            self.leftContainerViewWidthConstraint?.isActive = false
```

Comment:
From: https://developer.apple.com/documentation/uikit/uiview/1622659-removeconstraint

> When developing for iOS 8.0 or later, set the constraint’s isActive property to false instead of calling the removeConstraint(_:) method directly. The isActive property automatically adds and removes the constraint from the correct view.